server/src/api/v1.py

```diff
@@ -146,6 +146,47 @@ def job_get_id(job_id):
     return job_data
 
 
+@v1.get("/job/search")
+@v1.input(schemas.JobSearchRequest, location="query")
+@v1.output(schemas.JobSearchResponse)
+def search_jobs(query_data):
+    """Search for jobs by tags"""
+    tags = query_data.get("tags")
+    match = request.args.get("match", "any")
+    states = request.args.getlist("state")
+
+    if match not in ["any", "all"]:
+        abort(400, "Invalid match mode")
+
+    query = {}
+    if tags and match == "all":
+        query["job_data.tags"] = {"$all": tags}
+    elif tags and match == "any":
+        query["job_data.tags"] = {"$in": tags}
```

Comment:
What if no "tags" were passed. Since match defaults to any this would create the query `{"$in": []}` I assume? then would that not return anything?

Reply:
no, because if tags is empty, then it won't execute this section - we check both that tags contains something and that match==any before doing this.

Reply:
Oh yeah I missed that :sweat_smile: